autorag_research/orm/service/retrieval_pipeline.py

```diff
@@ -0,0 +1,238 @@
+        from autorag_research.orm.service import RetrievalPipelineService
+        from autorag_research.nodes.retrieval.bm25 import BM25Module
+
+        # Initialize BM25 module
+        bm25 = BM25Module(index_path="/path/to/index")
+
+        # Create service
+        service = RetrievalPipelineService(session_factory)
+
+        # Create pipeline
+        pipeline_id = service.create_pipeline(
+            name="bm25_baseline",
+            config={"type": "bm25", "index_path": "/path/to/index"},
+        )
+
+        # Run pipeline with BM25 retrieval function
+        results = service.run(
+            retrieval_func=bm25.run,
+            pipeline_id=pipeline_id,
+            metric_id=1,
+            top_k=10,
+        )
+        ```
+    """
+
+    def __init__(
+        self,
+        session_factory: sessionmaker[Session],
+        schema: Any | None = None,
+    ):
+        """Initialize retrieval pipeline service.
+
+        Args:
+            session_factory: SQLAlchemy sessionmaker for database connections.
+            schema: Schema namespace from create_schema(). If None, uses default schema.
+        """
+        self.session_factory = session_factory
+        self._schema = schema
+
+    def _get_schema_classes(self) -> dict[str, type]:
+        """Get schema classes from the schema namespace."""
+        if self._schema is not None:
+            return {
+                "Query": self._schema.Query,
+                "Pipeline": self._schema.Pipeline,
+                "ChunkRetrievedResult": self._schema.ChunkRetrievedResult,
+            }
+        from autorag_research.orm.schema import ChunkRetrievedResult, Pipeline, Query
+
+        return {
+            "Query": Query,
+            "Pipeline": Pipeline,
+            "ChunkRetrievedResult": ChunkRetrievedResult,
+        }
+
+    @contextmanager
+    def _session_scope(self):
+        """Provide a transactional scope around a series of operations."""
+        session = self.session_factory()
+        try:
+            yield session
+            session.commit()
+        except Exception:
+            session.rollback()
+            raise
+        finally:
+            session.close()
+
+    def _get_query_repo(self, session: Session) -> QueryRepository:
+        """Get Query repository for the given session."""
+        classes = self._get_schema_classes()
+        return QueryRepository(session, classes["Query"])
+
+    def _get_pipeline_repo(self, session: Session) -> PipelineRepository:
+        """Get Pipeline repository for the given session."""
+        return PipelineRepository(session)
+
+    def _get_chunk_retrieved_result_repo(self, session: Session) -> ChunkRetrievedResultRepository:
+        """Get ChunkRetrievedResult repository for the given session."""
+        classes = self._get_schema_classes()
+        return ChunkRetrievedResultRepository(session, classes["ChunkRetrievedResult"])
+
+    def create_pipeline(self, name: str, config: dict) -> int:
```

Comment:
I think the `create_pipeline` name is a little bit confused with the real `Pipeline` object.
How about 'add_pipeline' or any other name? Suggestions are needed.